src/dune_rules/lib.mli

```diff
@@ -100,8 +100,20 @@ module DB : sig
     val not_found : t
     val found : Lib_info.external_ -> t
     val to_dyn : t Dyn.builder
-    val redirect : db -> Loc.t * Lib_name.t -> t
+    val redirect : db -> Lib_id.t -> t
     val redirect_in_the_same_db : Loc.t * Lib_name.t -> t
+
+    module With_multiple_results : sig
+      type resolve_result := t
+
+      type t = private
+        | Resolve_result of resolve_result
+        | Multiple_results of resolve_result list
```

Comment:
You can go further. Given that `Resolve_result x = Multiple_results [ x ]`, there's no need for the individual constructors anymore. You can just have:

```
type result
type resolve_result = result list
```

Reply:
indeed. just did that in [b6ddab6](https://github.com/ocaml/dune/pull/10307/commits/b6ddab6dc5c20915d5987508865ac60a9e9860d8)